### ISSUE TITLE
[Metrics] Add Image Gradient

### ISSUE THREAD

**Issue by pranjaldatta** [PyTorchLightning/pytorch-lightning#4763]
## 🚀 Feature

Implement Image-Gradients for PT Lightning.

### Motivation

 Recently I was working on a vanilla PT implementation of the [DenseDepth paper](https://arxiv.org/abs/1812.11941). They happen to use a *DepthLoss* as one of their loss functions. Incidentally, DepthLoss is based on calculating *Image gradients* between two images, native implementation of which doesn't exist in PT! Including Image Gradient as a metric would make life easier for devs working on similar papers/projects. Also often metrics like ssim and image gradients are used in conjunction (as in this paper)

### Pitch

Make Image-gradients available as a metric out-of-box in PT Lightning

### Alternatives

Alternative solutions often include custom implementation of these metrics, thus increasing the development overhead.

**Comment by tchaton** [comment]
Hey @pranjaldatta,

Would you like to create a PR with the `Image gradients` metric implementation ?

Best regards,
Thomas Chaton.

**Comment by pranjaldatta** [comment]
@tchaton Yes! I am currently working on an implementation. Will make a PR asap.

**Comment by pranjaldatta** [comment]
Hey @tchaton, while working on this issue I ran into a doubt (pardon me if it's a naive thing to ask), 
So after adding the necessary image-gradient code and tests to metrics/functional, do I have to implement a class to in metrics/regression?

**Comment by SkafteNicki** [comment]
@pranjaldatta can this be closed?

**Comment by pranjaldatta** [comment]
Sure!
